addons/vol-slider/module.js

```diff
@@ -1,40 +1,43 @@
-const muteIcon = "/static/assets/e21225ab4b675bc61eed30cfb510c288.svg";
-const quietIcon = "/static/assets/3547fa1f2678a483a19f46852f36b426.svg";
-const loudIcon = "/static/assets/b2c44c738c9cbc1a99cd6edfd0c2b85b.svg";
+// Volumes in this file are always in 0-1.
 
 let vm;
-let vmVol;
-let defVol = 1;
-
-export const setVol = (v) => {
-  vmVol.value = v;
-  let slider = document.getElementById("sa-vol-slider");
-  let icon = document.getElementById("sa-vol-icon");
-  if (!slider) return;
-  slider.value = v;
-  if (v == 0) {
-    icon.src = muteIcon;
-  } else if (v < 0.5) {
-    icon.src = quietIcon;
-  } else {
-    icon.src = loudIcon;
-  }
-};
+/** @type {GainNode} */
+let gainNode;
+let unmuteVolume = 1;
+const callbacks = [];
 
-export const setDefVol = (v) => {
-  defVol = v;
+export const setVolume = (newVolume) => {
+  gainNode.value = newVolume;
+  callbacks.forEach((i) => i());
 };
 
-export const getDefVol = () => {
-  return defVol;
+export const getVolume = () => {
+  return gainNode.value;
 };
 
 export const isMuted = () => {
-  return vmVol.value === 0;
+  return getVolume() === 0;
+};
+
+export const setUnmutedVolume = (newUnmuteVolume) => {
+  unmuteVolume = newUnmuteVolume;
+};
+
+export const setMuted = (newMuted) => {
+  if (newMuted) {
+    setUnmutedVolume(getVolume());
+    setVolume(0);
+  } else {
+    setVolume(unmuteVolume);
+  }
+};
+
+export const onVolumeChanged = (callback) => {
+  callbacks.push(callback);
 };
 
 export const setup = (_vm) => {
   if (vm) return;
   vm = _vm;
-  vmVol = vm.runtime.audioEngine.inputNode.gain;
+  gainNode = vm.runtime.audioEngine.inputNode.gain;
```

Comment:
> Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'inputNode')

I'm not sure why this is happening now, but not before.

Reply:
was my bug introduced by this PR, should be fixed now